pkg/generate/app/imagelookup.go

```diff
@@ -79,6 +167,85 @@ func (r DockerRegistryResolver) Resolve(value string) (*ComponentMatch, error) {
 	}, nil
 }
 
+func descriptionFor(image *imageapi.DockerImage, value, from string) string {
+	shortID := image.ID
+	if len(shortID) > 7 {
+		shortID = shortID[:7]
+	}
+	parts := []string{fmt.Sprintf("Docker image %q", value), shortID, fmt.Sprintf("from %s", from)}
+	if image.Size > 0 {
+		mb := float64(image.Size) / float64(1024*1024)
+		parts = append(parts, fmt.Sprintf("%f", mb))
+	}
+	if len(image.Author) > 0 {
+		parts = append(parts, fmt.Sprintf("author %s", image.Author))
+	}
+	if len(image.Comment) > 0 {
+		parts = append(parts, image.Comment)
+	}
+	return strings.Join(parts, ", ")
+}
+
+func partialScorer(a, b string, prefix bool, partial, none float32) (bool, float32) {
+	switch {
+	case len(a) == 0 && len(b) != 0, len(a) != 0 && len(b) == 0:
+		return true, partial
+	case a != b:
+		if prefix {
+			if strings.HasPrefix(a, b) || strings.HasPrefix(b, a) {
+				return true, partial
+			}
+		}
+		return false, none
+	default:
+		return true, 0.0
+	}
+}
+
+func matchTag(image docker.APIImages, value, registry, namespace, name, tag string) []*ComponentMatch {
+	if len(tag) == 0 {
+		tag = "latest"
+	}
+	matches := []*ComponentMatch{}
+	for _, s := range image.RepoTags {
+		if value == s {
+			matches = append(matches, &ComponentMatch{
+				Value: s,
+				Score: 0.0,
+			})
+			continue
+		}
+		iRegistry, iNamespace, iName, iTag, err := imageapi.SplitDockerPullSpec(s)
+		if err != nil {
+			continue
+		}
+		if len(iTag) == 0 {
+			iTag = "latest"
+		}
+		match := &ComponentMatch{}
+		ok, score := partialScorer(name, iName, true, 0.5, 1.0)
```

Comment:
It does, in that if there's no relevant match you continue.  However, since these are additive all matches with name should be sorted before the others... I think...